koku/api/report/queries.py

```diff
@@ -181,11 +182,28 @@ def add(self, query_filter=None, table=None, field=None, operation=None, paramet
 
     def compose(self):
         """Compose filters into a dict for submitting to Django's ORM."""
-        out = {}
+        composed_query = None
+        compose_dict = {}
         for filt in self._filters:
-            if filt.parameter is not None:
-                out.update(filt.composed_dict())
-        return out
+            filt_key = filt.composed_query_string()
+            if compose_dict.get(filt_key) is None:
+                compose_dict[filt_key] = [filt]
+            else:
+                compose_dict[filt_key].append(filt)
+
+        for _, filter_list in compose_dict.items():
```

Comment:
If you're throwing away the key anyway, then just use `.values()` instead of `.items()`.